book_price/models/account_move_line.py

```diff
@@ -0,0 +1,24 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    book_price = fields.Float(string='Book Price', compute='_compute_book_price', store=True)
+
+    @api.depends('product_id', 'quantity', 'move_id.invoice_origin')
+    def _compute_book_price(self):
+        for line in self:
+            line.book_price = 0.0 
+            if not (line.product_id and line.move_id.invoice_origin):
+                continue
+
+            sale_order = self.env['sale.order'].search([('name', "=", line.move_id.invoice_origin)], limit=1)
+            if sale_order:
+                sale_order_line = self.env['sale.order.line'].search([
+                    ('order_id', "=", sale_order.id),
+                    ('product_id', "=", line.product_id.id)
+                ], limit=1)
```

Comment:
```suggestion
            line.book_price = 0.0
            sale_order_line = line.sale_line_ids.filtered(lambda sol: sol.product_id == line.product_id)[:1]
```
We already have a relation between `account.move.line` and `sale.order.line`. Maybe, can we just directly use that? What do you say? :thinking: 
In both approaches, if there is same product in a sale order but in 2 or more different SOL, what would be the scenario?

Reply:
-Yes, we already have a relation between account.move.line and sale.order.line through sale_line_ids, so directly using it is the better approach instead of manually searching for sale.order and sale.order.line.
-In the current search() approach, if a sale order has multiple lines for the same product, only the first matching SOL is considered (limit=1), which may not always be correct.
By filtering sale_line_ids, we ensure we are selecting the correct sale order line based on product_id, quantity, and product_uom.

Reply:
It would be better if we pass this field value in `_prepare_invoice_line` method instead of finding the related sale.order.line.